src/TaskManager.Infrastructure/Repositories/Repository.cs

```diff
@@ -1,50 +1,63 @@
 ﻿using Microsoft.EntityFrameworkCore;
-using TaskManager.Core.Interfaces;
+using System.Linq.Expressions;
+using TaskManager.Core.Interfaces.Data;
 
 namespace TaskManager.Infrastructure.Repositories;
 
 internal class Repository<TModel> : IRepository<TModel> where TModel : class
 {
     protected readonly DbContext _context;
-    private readonly DbSet<TModel> _dbSet;
-    public Repository(DbContext context) 
+    protected readonly DbSet<TModel> _dbSet;
+
+    public Repository(DbContext context)
     {
         _context = context ?? throw new ArgumentNullException(nameof(context));
         _dbSet = _context.Set<TModel>();
     }
 
-    public Task<List<TModel>> GetAllAsync()
+    public Task<List<TModel>> GetAllAsync(CancellationToken? cancellationToken)
     {
-        return _dbSet.ToListAsync();
+        return _dbSet.ToListAsync(cancellationToken ?? CancellationToken.None);
     }
 
-    public Task<TModel> GetByIdAsync(Guid id)
+    public Task<TModel> GetByIdAsync(Guid id, CancellationToken? cancellationToken)
     {
-        return _dbSet.FindAsync(id).AsTask();
+        return _dbSet.FindAsync(id, cancellationToken ?? CancellationToken.None).AsTask();
     }
 
-    public async Task AddAsync(TModel model)
+    public async Task<TModel> AddAsync(TModel model, CancellationToken? cancellationToken)
     {
         await _dbSet.AddAsync(model);
-      
-        await _context.SaveChangesAsync();
+
+        await _context.SaveChangesAsync(cancellationToken ?? CancellationToken.None);
+
+        return model;
     }
 
-    public Task UpdateAsync(TModel model)
+    public async Task<bool> UpdateAsync(TModel model, CancellationToken? cancellationToken)
     {
         _context.Entry(model).State = EntityState.Modified;
 
-        return _context.SaveChangesAsync();
+        var result = await _context.SaveChangesAsync(cancellationToken ?? CancellationToken.None);
+
+        return result > 0;
     }
 
-    public async Task DeleteAsync(Guid id)
+    public async Task DeleteAsync(Guid id, CancellationToken? cancellationToken)
     {
-        var entity = await _dbSet.FindAsync(id);
+        var entity = await _dbSet.FindAsync(id, cancellationToken ?? CancellationToken.None);
         if (entity != null)
         {
-            _dbSet.Remove(entity);
-            
+            typeof(TModel).GetProperty("IsDeleted")?.SetValue(entity, true);
+
+            _context.Entry(entity).State = EntityState.Modified;
+
             await _context.SaveChangesAsync();
         }
     }
+
+    public async Task<IEnumerable<TModel>> FindAsync(Expression<Func<TModel, bool>> predicate, CancellationToken? cancellationToken)
+    {
+        return await _dbSet.Where(predicate).ToListAsync(cancellationToken ?? CancellationToken.None);
+    }
```

Comment:
```suggestion
    public Task<List<TModel>> FindAsync(Expression<Func<TModel, bool>> predicate, CancellationToken cancellationToken = default)
    {
        return _dbSet.Where(predicate).ToListAsync(cancellationToken);
    }
```

Я это имелл ввиду, когда писал про ctoken :) Налабл его делать не нужно. Вообще стремимся избегать налов в коде

Reply:
и все таки почему ты не хочешь использовать CancellationToken cancellationToken = default?🤔
Это, конечно, дело вкуса, но дает возможность не использовать его там, где его нет и выглядит красивей

Reply:
я просто не знал про такое